post.md

```diff
@@ -0,0 +1,427 @@
+
+## Sample application
+
+This article presumes you're acquainted with setting up and running Next.js applications.
+
+To view the code referenced in this article, [checkout the repo](https://github.com/CodingZeal/nextjs-example-confirm-server-action).
+
+This article and accompanying example application utilize Next.js version 13.5.4. Bear in mind that server actions remain in the experimental phase and could evolve.
+
+## A Glimpse at Server Actions
+
+Next.js's server actions empower developers to execute server-side operations directly from the client without the hassle of wiring up HTTP client code. As the Next.js documentation succinctly puts it:
+
+> With Server Actions, you don't need to manually create API endpoints. Instead, you define asynchronous server functions that can be called directly from your components.
+
+### Required Next.js Configurations
+
+To leverage server actions, the experimental feature flag must be enabled in your Next.js configuration:
+
+```ts
+module.exports = {
+  experimental: {
+    serverActions: true,
+  },
+};
+```
+
+### Basic Server Action Example
+
+Consider the following rudimentary example that sends form data from the client to the server action and then renders to the server console. Note, the server action is set as the form's `action`:
+
+```tsx
+export default function SomePage() {
+  async function exampleServerFunction(formData: FormData) {
+    "use server";
+
+    console.log("formData", Array.from(formData));
+  }
+
+  return (
+    <div>
+      <form action={exampleServerFunction}>
+        <div>
+          <label htmlFor="name">Name: </label>
+          <input type="text" name="name" />
+        </div>
+
+        <button type="submit">Submit</button>
+      </form>
+    </div>
+  );
+}
+```
+
+## Using the experimental `useFormStatus` hook for user feedback
+
+At its core, `useFormStatus` tells us the current state of our form's server action, allowing for dynamic UI updates based on server action outcomes.
+
+The hook currently returns a `pending` property, which is set to true during the form submission process. This can be incredibly useful for preventing multiple submissions and giving users feedback that the form submission is still in process.
+
+Before we continue here is the server action we are calling:
+
+```ts
+"use server";
+
+export async function deleteSomething(params: FormData) {
+  const id = params.get("id");
+  const type = params.get("type");
+
+  console.log(`[${type}] Deleting ${id}...`);
+
+  // Simulate a long-running operation
+  await new Promise((resolve) => setTimeout(resolve, 2000));
+}
+```
+
+The following example shows how to use `useFormStatus` to disable the submit button while the form is submitting:
+
+- this component contains a form with a server action bound to its action attribute
+- the form is submitted via a child component: `<ServerActionButton />`
```

Comment:
I'd probably weave these bullets into the previous paragraph. You can say something like, "The following example shows how to use `useFormStatus` to disable the submit button while the form is submitting. Keep in mind that this component contains a form with a server action bound to its action attribute. Also, the form is submitted via a child component: `<ServerActionButton />`"

Reply:
Going to stick with my bullets on this one section